src/components/ui/MultipleSelect/MultipleSelect.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import {Select} from 'antd'
+import {SelectProps} from 'antd/lib/select'
+import {WidgetField} from '../../../interfaces/widget'
+import {MultivalueSingleValue} from '../../../interfaces/data'
+
+type AdditionalAntdSelectProps = Partial<Omit<SelectProps, 'onChange' | 'value' | 'className' | 'disabled' | 'placeholder'>>
+
+export interface MultipleSelectProps extends AdditionalAntdSelectProps {
+    value: MultivalueSingleValue[],
+    values: Array<{ value: string }>,
+    onChange?: (value: MultivalueSingleValue[]) => void
+    // field common props
+    meta: WidgetField,
+    cursor: string,
+    widgetName: string,
+    className: string,
+    metaError: string,
+    disabled: boolean,
+    placeholder: string,
+    readOnly: boolean,
+    backgroundColor: string,
+    onDrillDown: () => void,
+}
+
+const MultipleSelect: React.FunctionComponent<MultipleSelectProps> = (props) => {
+    const {
+        value,
+        values,
+        onChange,
+        // field common props
+        meta,
+        cursor,
+        widgetName,
+        className,
+        metaError,
+        disabled,
+        placeholder,
+        readOnly,
+        backgroundColor,
+        onDrillDown,
+        ...rest
+    } = props
+
+    const choiceValues = React.useMemo(
+        () => {
+            return values?.map((item) => {
+                return (
+                    <Select.Option key={item.value} label={item.value}>
+                        <span>{item.value}</span>
+                    </Select.Option>
+                )
+            })
+        },
+        [values]
+    )
+
+    const handleOnChange = React.useCallback(
+        (v: string[]) => {
+            const result: MultivalueSingleValue[] = []
+            v.map(item => result.push({id: item, value: item}))
```

Comment:
It's either
```tsx
const result = v.map(item => ({ id: item, value: item }))
```
or
```tsx
const result: MultivalueSingleValue[] = []
v.forEach(item => {
    result.push({ id: item, value: item })
})
```

There is no point in mapping if the result is not assigned to something: essentially you've mapped `v` to a new array `[1, 2, 3, ..., v.length - 1]` and then dropped it.